### ISSUE TITLE
[VSCode] Check identifier consistency in properties dialogs

### ISSUE THREAD

**Issue by joelvonrotz** [quarto-dev/quarto#199]
Hello Quarto Devs

I've been working a lot with the Visual Editor in VSCode and love it a lot so far. Overall everything works fine except for some small glitches here and there (sometimes it reverts small amounts of freshly written letters), but I'm focusing on a different thing today:

Recently I noticed that the properties for the components are a tad bit inconsistent. For example the properties dialog for equations and figures both have different ways to set the identifier (figure requiring a `#` and the equation doesn't). And there is no properties dialog for the tables.

![grafik](https://github.com/quarto-dev/quarto/assets/41155337/2572b322-60ee-4684-b3ac-c819dbcd331c)

![grafik](https://github.com/quarto-dev/quarto/assets/41155337/b64b1e8d-5a6d-4ef4-aa53-50bef784bd8f)

Is this a limitation of the prosemirror editor?

But else it's a fine tool, good stuff!!

**Comment by jjallaire** [comment]
No, this is just a bug! (thanks for reporting). Fixed here: https://github.com/quarto-dev/quarto/commit/fe5c95e7fe9993d5838c4d277f2b190201e97dce  (should be available within a few days)

w/r/t to the reverting of freshly written letters, it's possible that this occurs when you have a source editor also open onto the same file (there is complex logic that deals with synchronizing concurrent edits to both and sometimes it needs to back out changes to be on the safe side). I'd expect to see this if focus changes right before or after the behavior. Is it possible that this is what you are observing?

**Comment by joelvonrotz** [comment]
Thanks for the response and sorry for my rather late response.
With the conflicting, yeah that seems to be the case. I'm now a bit more careful when editing docs in the visual editor (seems to happen more there).

Also, I've fought about the cross referencing process a bit. What do you think of only setting the name of identifier, while automatically attaching the corresponding prefix during rendering. As an example for the figure dialog, you'd see th prefix `#fix-` and an input field to name the figure:

![grafik](https://github.com/quarto-dev/quarto/assets/41155337/9e6bbc93-3984-4042-83e4-e620df0ea8cf)

**Comment by jjallaire** [comment]
@joelvonrotz that's an interesting suggestion. One reason that the `#fig-` prefix is there is to designate that this is a cross-reference able figure (as opposed to just a div or image with an ID). This implies for example that it will appear in the list of figures and a caption is _required_. I wonder if we could just solve this with user-interface (e.g. designate that its a figure using the UI and in that case auto inject the `#fig-` prefix.

**Comment by joelvonrotz** [comment]
Thanks for giving your opinion on it. I dragged some people in a group project into using Quarto as our documentation tool and it was just something I noticed. They sometimes forgot to prefix the identifier with `#fig-`.

How the cross referencing system is implemented is fine, but I guess the step of prefixing each identifier is a bit reduntant for the user and could be automated. Of course the user needs some kind of visual indication, that something will be prefixed (like in the image I posted previously).

Also I don't know if it impacts the flexibility of writing whatever into the identifier dialog.

**Comment by jjallaire** [comment]
I agree this could be automated, especially within the context of a tool where the user's semantic intent is clear (e.g. "Insert Figure" or "Insert Table" / "Insert Equation" with a numbering option set to true. In some ways you are working at the protocol level (where intent is encoded in the `#fig-` prefix) but as you point out that's a bit too fiddly to not cause confusion/problems. 

One other consideration is that we use the `@` symbol as the prefix for both citations and cross references -- this means we need to clearly disambiguate intent and the prefix provides some type information that helps with this. If the user provides an identifier (e.g. `#big-elephant`) that we we auto-prefix to `#fig-big-elephant` then they need to know at the time of referencing to include the prefix (and they might not know to since we auto-inserted it). Again, this can be solved at the level of authoring tools (and we will certainly do that!)
